cmd/chaos/chaos.go

```diff
@@ -1,8 +1,86 @@
 package main
 
-import "github.com/projectdiscovery/chaos-client/internal/runner"
+import (
+	"fmt"
+	"github.com/alecthomas/kong"
+	"github.com/projectdiscovery/chaos-client/internal"
+	"github.com/projectdiscovery/chaos-client/internal/subdomains"
+	"github.com/projectdiscovery/gologger"
+	"github.com/projectdiscovery/gologger/levels"
+	updateutils "github.com/projectdiscovery/utils/update"
+	"os"
+)
+
+var cli struct {
+	Key                string `help:"Chaos key for API" short:"k"`
+	Domain             string `help:"Domain to search for subdomains" short:"d"`
+	Silent             bool   `help:"Make the output silent"`
+	Output             string `help:"File to write output to (optional)" short:"o"`
+	Version            bool   `help:"Show version of chaos"`
+	Verbose            bool   `help:"Verbose" short:"v"`
+	Update             bool   `help:"update Chaos to latest version" aliases:"up"`
+	DisableUpdateCheck bool   `help:"disable automatic Chaos update check" aliases:"duc"`
+	Subdomains         struct {
+		Domain string `arg:"" help:"Domain."`
+		Count  bool   `help:"Show statistics for the specified domain"`
+		DL     string `help:"File containing domains to search for subdomains (optional)" aliases:"dL"`
+		JSON   bool   `help:"Print output as json"`
+	} `cmd:"" help:"List subdomains"`
+
+	DNS struct {
+		Paths []string `arg:"" optional:"" help:"Paths to list." type:"path"`
+	} `cmd:"" help:"Get DNS record"`
+}
 
 func main() {
-	opts := runner.ParseOptions()
-	runner.RunEnumeration(opts)
+	ctx := kong.Parse(&cli,
+		kong.Name("Chaos"),
+		kong.Description("Chaos client"),
+		kong.UsageOnError(),
+		kong.ConfigureHelp(kong.HelpOptions{
+			Compact: true,
+			Summary: true,
+		}))
+
+	if cli.Silent {
+		gologger.DefaultLogger.SetMaxLevel(levels.LevelSilent)
+	}
+	internal.ShowBanner()
+
+	if cli.Version {
+		gologger.Info().Msgf("Current Version: %s\n", internal.Version)
+		os.Exit(0)
+	}
+
+	if !cli.DisableUpdateCheck {
+		latestVersion, err := updateutils.GetVersionCheckCallback("chaos-client")()
+		if err != nil {
+			if cli.Verbose {
+				gologger.Error().Msgf("chaos version check failed: %v", err.Error())
+			}
+		} else {
+			gologger.Info().Msgf("Current chaos version %v %v", internal.Version, updateutils.GetVersionDescription(internal.Version, latestVersion))
+		}
+	}
+
+	switch ctx.Command() {
+	case "subdomains <domain>":
```

Comment:
Supporting only this particular case can result in a breaking change. We have two options to consider. One, we can proceed with the changes and include a warning when releasing the tool. Alternatively, we can support the old CLI flags, mark them deprecated, and remove them in future versions.

Reply:
Sorry I forgot to set Draft to PR. Work in progress

Reply:
@dogancanbakir Let's just release V2? The changes are breaking and it is not so trivial to support backward compatibility:
1. The current version operates with wrong argument format (it uses a single `-` for both short and long flags, whereas standard — two hyphens for long flags)
2. Because of this, https://github.com/alecthomas/kong (our new CLI tool) doesn't support single hyphen long flags